src/metadataGeneration/typeResolver.ts

```diff
@@ -199,7 +204,11 @@ export class TypeResolver {
 
     if (extractEnum) {
       const enums = enumDeclaration.members.map((member: any, index) => {
-        return getEnumValue(member) || String(index);
+        const enumValue = getEnumValue(member);
+        if (enumValue !== 0 && !enumValue) {
```

Comment:
Can't we check for `undefined` here?

Reply:
He’s saying that it can’t be null, undefined or an empty string but that it can be 0. If he just checked for not undefined then he would lose the empty string check and also 0 would be in appropriately flagged.